### PR TITLE
Remove reliance on setting global system env vars in tests

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

**Comment by trask** [comment]
hey @lmolkova @srnagar - both @heyams and @jeanbisutti are out for a few weeks, would one of you be able to review this PR for me?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java [hunk 1]
```diff
@@ -302,9 +303,13 @@ private TelemetryItemExporter initExporterBuilder() {
                         TempDirs.getSubDir(tempDir, "telemetry"),
                         pipeline,
                         LocalStorageStats.noop(),
-                        false));
+                        false),
+                    // TODO (trask) pass in autoconfigure's ConfigProperties after converting this to autoconfigure
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java:307]: It'd be great to understand how otel and Azure SDK configurations could coexist.

You already have it in the builder https://github.com/Azure/azure-sdk-for-java/blob/7c644b36ed10cb9d2af0c89b4967d5df255a1187/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java#L166


Check out `com.azure.core.util.Configuration` and example - 
https://github.com/Azure/azure-sdk-for-java/blob/654650746b7294af2f81457cde7db9afbbee76e2/sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/ProxyOptions.java#L56

example usage

https://github.com/Azure/azure-sdk-for-java/blob/654650746b7294af2f81457cde7db9afbbee76e2/sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/ProxyOptionsTests.java#L133

it supports env vars, system properties, can be extended to support yaml files. 

I wonder if we can make some reasonable story (e.g. az Configuration can be built from source that reads otel properties?

**Review by lmolkova — APPROVED** [review]
LGTM, left a comment on future configuration APIs

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java (hunk 1 above)
Comment (trask) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java:307]: this is a great idea! I'll go explore it in #36230

**Pull request merged**